### ISSUE TITLE
Provide a way to set retro passwords

### ISSUE THREAD

**Issue by marciobarrios** [retro-tool/retro-tool#6]
As retrotool.app is now available to the public, and retros might contain private or strategic information about the company, we would need to provide an option to add a password to the retro when it's created.

When a retro has a password, access should be restricted unless the user enters the password.

**Comment by uesteibar** [comment]
the discussion in https://github.com/retro-tool/retro-tool/issues/20 is also relevant to this, as we'd need some kind of config panel to set this

**Comment by uesteibar** [comment]
Can we assume that the password can be updated at any time, and that anyone with access to the retro (and thus has the password) can update the password? Since we don't have anything like _users_, it would be complicated to do otherwise.

**Comment by rudeayelo** [comment]
Yes, anyone with the password has the ability to change it, like we're god-level communists

**Comment by uesteibar** [comment]
we've discussed about the UX to _update_ the password for a retro (or set it the first time). What about entering the password to access the retro?

**Comment by rudeayelo** [comment]
I propose an invisible input in a black screen (focus is just assumed to have been captured), then when the user writes the password and taps "return" there's two possible outcomes:

- ACCESS DENIED (big uppercase red letters) then back to the black screen
- ACCESS GRANTED (big uppercase green letters) then load the retro after a 3 seconds delay, we need to make sure the user knows the outcome was positive

**Comment by uesteibar** [comment]
@rudeayelo now seriously, just a popup with the password input if the request for the retro returns a 401 error would do, right?

**Comment by rudeayelo** [comment]
I'd say so, yes
